README.md

```diff
@@ -126,18 +136,49 @@ Yes, but we already have to agree on functions, so this is not hard. The table e
+Its syntactic similarity to HTTP headers and [support for](https://datatracker.ietf.org/doc/html/rfc6920#section-3.1), MIME content-types makes it potentially useful for web use-cases, but use-cases are not constrained by URI scheme, only hinted at by the specification in sections 3 through 7.
+
+#### Translation from multihash to named-information hash
+
+Translating from a bare, binary multihash (i.e., a hash value in `unsigned_varint`, a.k.a. ULEB128 format) to a named-information hash in binary format is fairly easy to do insofar as a generic tag for self-describing multihashes was proposed to the [NIH registry][] by [Appendix B](https://www.ietf.org/archive/id/draft-multiformats-multihash-03.html#appendix-D.2) in the 2021 [multihash internet draft](https://www.ietf.org/archive/id/draft-multiformats-multihash-03.html):
+
+1. Strip the prefix bytes from the hash value and use the prefix bytes to identity the hash function used from the [Multicodec][] table
+2. If multihash prefix corresponds to any tags in the [NIH registry][]:
+  1. translate multicodec tag to NIH tag, i.e., if `0x12` (`sha2-256`) in `multicodec` registry, then `0x01` (`sha256`) in `named-information` registry
+  2. transcode the hash value from ULEB128 to standard MSB binary
+  3. (for binary form:) reattach new prefix to transcoded hash value
+  4. (for ASCII form:) convert prefix to URL format, i.e., `ni:///sha-256;` for `0x01`, and reattach to base64-encoded transcoded hash value
+3. If multihash prefix does NOT map cleanly to a registered value in [NIH registry][]:
+   1. (for binary form:) prefix existing binary multihash with `0x42` to designate that what follows is a multicodec prefix followed by an ULEB128 hash value.
+   2. (for ASCII form:) convert the `0x42` prefix to URL format, i.e., `ni:///mh;` and then append a base64url, no-padding encoding of the entire binary multihash with prefix (and _without_ adding the additional base-64-url-no-padding prefix, `u`, if using a [multibase][] library for this base-encoding).
+
+Note that raw multihashes (i.e. multihashes directly taken from hashing inputs) are not commonly used in IPFS implementations, since inputs are usually broken up into an intermediary form before being hashed.
+Only "single-block" CIDs, which are directly produced from inputs without file-system conversion, can be converted as described above; these are usually used for blobs below a certain size, typically using `raw` or `json` or other non-IPLD tags to mark their referents as only one-layer deep.
+To translate between CIDs that dereference to an IPLD graph or other recursive structure, you must first reconstruct the inputs and re-encode a new CID using `raw` codec and no chunking structure, indirection, recursion, or outer envelope.
```

Comment:
Actually that's a different question-- the questions I got at IETF were "doesn't IETF already have a self-describing hash notation?" (i.e., named information hash), and more generally, a question I got of all the multiformats family, "how does this new set of formats translate to and co-exist with existing IETF formats?" (which is why DataURIs is next on my list of prior art to work out translation for).  I'm trying to situate the IPFS CID (and IPLD) as one consumer of many and map out the translation so that any NIH could be deterministically mapped to one multihash and vice versa.  Because otherwise applying multihash as an RFC is telling people to deprecate NIH, which has a higher burden of proof than just offering multihash as an optional upgrade path for NIH systems 😅 

You make the very good point that perhaps a translation algorithm is too big for the multihash `README.md` and could be defined separately in, idunno, an IPIP or at least a separate file elsewhere in this org.  I am open to ideas on how to elaborate this into an entire spec and where to put it, if people prefer that!

But wherever the "original" sits, the intention is to include an appendix in the next version of the IETF RFC for multihash, since _there_ is where the above questions arise and should be answered.

Reply:
could you just delete these three lines? does it really add much value here?

Reply:
^ That works, too, I agree it's kind of overkill _here_ (maybe not in if it were, say, an IPIP)

Reply:
Removed [in this commit](https://github.com/multiformats/multihash/pull/164/commits/a0b23da567d931724bca2018bd6255314a6651db)

Reply:
fine by me, @aschmahmann does this relieve your concerns enough?